src/react/ReportFooter/ReportFooter.tsx

```diff
@@ -0,0 +1,21 @@
+import React, { FC } from 'react';
+
+import StatusKey from './components/StatusKey';
+import AutoSchedulerFooter from './components/AutoSchedulerFooter';
+import { usePrimaryReportType } from '../ReportControls/hooks/usePrimaryReportType';
+
+const Footer: FC = () => {
+  const [primaryReportType] = usePrimaryReportType();
+
+  if (primaryReportType === 'start-due') {
+    return <StatusKey />;
+  }
+
+  if (primaryReportType === 'auto-scheduler') {
+    return <AutoSchedulerFooter />;
+  }
+
+  return null;
```

Comment:
qq: did we want to keep the map pattern we've been using?

Reply:
I'd like to. I've done a refactor, let me know what you think